python/hail/representation/pedigree.py

```diff
@@ -241,26 +259,28 @@ def filter_to(self, samples):
         return Pedigree._from_java(self._jrep.filterTo(jset(samples)))
 
     @handle_py4j
+    @typecheck_method(path=strlike)
     def write(self, path):
         """Write a .fam file to the given path.
 
         **Examples**
-        
+
         >>> ped = Pedigree.read('data/test.fam')
         >>> ped.write('out.fam')
-        
+
         **Notes**
 
         This method writes a `PLINK .fam file <https://www.cog-genomics.org/plink2/formats#fam>`_.
 
         .. caution::
-        
+
             Phenotype information is not preserved in the Pedigree data structure in Hail.
             Reading and writing a PLINK .fam file will result in loss of this information.
-            Use the key table method :py:meth:`~hail.KeyTable.import_fam` to manipulate this 
+            Use the key table method :py:meth:`~hail.KeyTable.import_fam` to manipulate this
             information.
 
-        :param str path: output path
+        :param path: output path
```

Comment:
If there's one type, we use this syntax:
```
:param str path: ...
```

If there are multiple possible types (or a :py:meth:...), we use this syntax:
```
:param path: ...
:type path: str or None
```